src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -225,10 +225,39 @@ async function signTx(req, res) {
 
   const partialTx = req.body.partial_tx;
   const signatures = req.body.signatures || [];
+  const { proposal_id: proposalId, version } = req.body.service || {};
 
   try {
     const tx = atomicSwapService.assembleTransaction(partialTx, signatures, req.wallet.storage);
 
+    // Updating the service, if a proposal id was informed
+    if (constants.SWAP_SERVICE_FEATURE_TOGGLE && proposalId) {
+      // First, validate if the proposal is already registered with this wallet
+      const listenedProposals = await atomicSwapService.getListenedProposals(req.walletId);
+      if (!listenedProposals.has(proposalId)) {
+        res.status(404);
+        res.send({
+          success: false,
+          error: 'Proposal is not registered. Register it first through [POST] /register/:proposalId',
```

Comment:
The URL is wrong, we have some prefix (`/wallet/atomic-swap/...`).
Also, should we have this URL on the error message?